### PR TITLE
Add podman support

### LINKED ISSUE #13617: [autoscaler] Add support for podman in Ray's autoscaler command line tool
As of now there is no good way to run the autoscaler with podman rather than docker, I tried setting 'docker' as an alias to podman in the initialization_commands section to no avail. 

Happy to share more details if necessary

### PR DESCRIPTION
**Pull request by jCrompton**
## Why are these changes needed?

This change is an attempt to support podman in addition to docker in the autoscaler CLI

## Related issue number
Closes #13617

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/_private/command_runner.py [hunk 1]
```diff
@@ -584,6 +584,9 @@ def __init__(self, docker_config, **common_args):
         self.docker_config = docker_config
         self.home_dir = None
         self.initialized = False
+        # Optionally use 'podman' instead of 'docker'
+        use_podman = docker_config.get("use_podman", False)
```
Comment (ijrsvt) [inline python/ray/autoscaler/_private/command_runner.py:588]: This looks good!, Can you make sure to add this to:
[ray-sechema.json](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/ray-schema.json).

**Review by ijrsvt — COMMENTED** [review]
Can you also modify [docker.py](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/docker.py)? It has some other calls to `docker exec` or `docker inspect`.

File: python/ray/autoscaler/_private/command_runner.py (hunk 1 above)
Comment (jCrompton) [reply python/ray/autoscaler/_private/command_runner.py:588]: Sure thing!

Comment (ijrsvt) [reply python/ray/autoscaler/_private/command_runner.py:588]: Looks good!

**Comment by ijrsvt** [comment]
@jCrompton are you still able to work on this PR?

**Comment by jCrompton** [comment]
Sorry I can no longer work on this, don't have the permission from my company :( sorry for the inconvenience. Hopefully will be able to soon

**Comment by ijrsvt** [comment]
> Sorry I can no longer work on this, don't have the permission from my company :( sorry for the inconvenience. Hopefully will be able to soon

@jCrompton No worries at all! I can finish this up :)

**Comment by jCrompton** [comment]
Thanks @ijrsvt !

File: python/ray/autoscaler/_private/command_runner.py [hunk 2]
```diff
@@ -668,40 +672,49 @@ def run_rsync_down(self, source, target, options=None):
...
+            if self.docker_cmd == "docker":
+                install_commands = [
+                    "curl -fsSL https://get.docker.com -o get-docker.sh",
+                    "sudo sh get-docker.sh", "sudo usermod -aG docker $USER",
+                    "sudo systemctl restart docker -f"
+                ]
+            else:
+                install_commands = [
+                    "sudo apt - get update",
+                    "sudo apt - get - y install podman"
```
Comment (ijrsvt) [inline python/ray/autoscaler/_private/command_runner.py:704]:
```suggestion
                    "sudo apt-get update",
                    "sudo apt-get - y install podman"
```

File: python/ray/autoscaler/_private/command_runner.py [hunk 3]
```diff
@@ -668,40 +672,49 @@ def run_rsync_down(self, source, target, options=None):
...
+            if self.docker_cmd == "docker":
+                install_commands = [
+                    "curl -fsSL https://get.docker.com -o get-docker.sh",
+                    "sudo sh get-docker.sh", "sudo usermod -aG docker $USER",
+                    "sudo systemctl restart docker -f"
+                ]
+            else:
+                install_commands = [
+                    "sudo apt-get update",
+                    "sudo apt-get - y install podman"
```
Comment (ijrsvt) [inline python/ray/autoscaler/_private/command_runner.py:704]:
```suggestion
                    "sudo apt-get update",
                    "sudo apt-get -y install podman"
```

File: python/ray/autoscaler/_private/command_runner.py [hunk 4]
```diff
@@ -799,12 +815,15 @@ def run_init(self, *, as_head, file_mounts, sync_run_yet):
...
-                self.run(f"docker stop {self.container_name}", run_env="host")
+                self.run(
+                    f"{self.docker_cmd} stop {self.container_name}",
+                    run_env="host")
 
         if (not container_running) or requires_re_init:
             # Get home directory
             image_env = self.ssh_command_runner.run(
-                "docker inspect -f '{{json .Config.Env}}' " + specific_image,
+                f"{self.docker_cmd} inspect -f '{{json .Config.Env}}' " +
```
Comment (ijrsvt) [inline python/ray/autoscaler/_private/command_runner.py:825]:
```suggestion
                f"{self.docker_cmd}" + "inspect -f '{{json .Config.Env}}' " +
```

File: python/ray/autoscaler/_private/command_runner.py [hunk 5]
```diff
@@ -846,7 +867,7 @@ def _configure_runtime(self):
             return []
 
         runtime_output = self.ssh_command_runner.run(
-            "docker info -f '{{.Runtimes}}' ",
+            f"{self.docker_cmd} info -f '{{.Runtimes}}' ",
```
Comment (ijrsvt) [inline python/ray/autoscaler/_private/command_runner.py:870]:
```suggestion
            f"{self.docker_cmd}" + "info -f '{{.Runtimes}}' ",
```

File: python/ray/autoscaler/ray-schema.json [hunk 6]
```diff
@@ -247,7 +247,12 @@
                     "type": "boolean",
                     "description": "disable Ray from automatically detecting /dev/shm size for the container",
                     "default": false
-                }
+                },
+              "use_podman"  :  {
+                "type": "boolean",
+                "description": "Use 'podman' command in place of 'docker'",
+                "default": false
+              }
```
Comment (simon-mo) [inline python/ray/autoscaler/ray-schema.json:255]:
```suggestion
                },
              "use_podman"  :  {
                "type": "boolean",
                "description": "Use 'podman' command in place of 'docker'",
                "default": false
                }
```

File: python/ray/autoscaler/_private/command_runner.py [hunk 7]
```diff
@@ -668,40 +672,49 @@ def run_rsync_down(self, source, target, options=None):
...
+            if self.docker_cmd == "docker":
+                install_commands = [
+                    "curl -fsSL https://get.docker.com -o get-docker.sh",
+                    "sudo sh get-docker.sh", "sudo usermod -aG docker $USER",
+                    "sudo systemctl restart docker -f"
+                ]
+            else:
+                install_commands = [
+                    "sudo apt-get update",
+                    "sudo apt-get -y install podman"
```
Comment (ijrsvt) [inline python/ray/autoscaler/_private/command_runner.py:704]:
```suggestion
                    "sudo apt-get update", "sudo apt-get -y install podman"
```

**Review by simon-mo — APPROVED** [review]

**Pull request merged**